src/components/icon/icon-close.spec.js

```diff
@@ -0,0 +1,399 @@
+      const wrapper = renderStyles({ disabled: true, bgTheme: 'business' });
+      assertStyleMatch(
+        {
+          color: baseTheme.icon.disabled
+        },
+        wrapper.toJSON()
+      );
+    });
+
+    describe.each(['error', 'info', 'business'])(
+      'when the background theme is %s',
+      (whiteIconBackground) => {
+        it('renders a white icon', () => {
+          const wrapper = renderStyles({ bgTheme: whiteIconBackground });
+          assertStyleMatch(
+            {
+              color: baseTheme.colors.white
+            },
+            wrapper.toJSON()
+          );
+        });
+      }
+    );
+
+    describe.each(['success', 'warning'])(
+      'when the background theme is %s',
+      (darkIconBackground) => {
+        it('renders a dark icon', () => {
+          const wrapper = renderStyles({ bgTheme: darkIconBackground });
+          assertStyleMatch(
+            {
+              color: baseTheme.icon.default
+            },
+            wrapper.toJSON()
+          );
+
+          assertStyleMatch(
+            {
+              color: baseTheme.icon.defaultHover
+            },
+            wrapper.toJSON(),
+            { modifier: ':hover' }
+          );
+        });
+      }
+    );
+
+    describe('when bgTheme is set to none', () => {
+      it('renders white icon when iconColor is set to onDarkBackground', () => {
+        const wrapper = renderStyles({ iconColor: 'on-dark-background', bgTheme: 'none' });
+        assertStyleMatch(
+          {
+            color: baseTheme.colors.white
+          },
+          wrapper.toJSON()
+        );
+      });
+
+      it('renders dark icon when iconColor is set to onLightBackground', () => {
+        const wrapper = renderStyles({ iconColor: 'on-light-background', bgTheme: 'none' });
+        assertStyleMatch(
+          {
+            color: baseTheme.icon.onLightBackground
+          },
+          wrapper.toJSON()
+        );
+
+        assertStyleMatch(
+          {
+            color: baseTheme.icon.onLightBackgroundHover
+          },
+          wrapper.toJSON(),
+          { modifier: ':hover' }
+        );
+      });
+
+      describe('fontSize is "large" and bgSize is "small"', () => {
+        it('sets the height and width style properties of bgSize to same as "large"', () => {
+          const wrapper = renderStyles({ fontSize: 'large', bgTheme: 'foo', bgSize: 'small' });
+          assertStyleMatch(
+            {
+              height: '40px',
+              width: '40px'
+            },
+            wrapper.toJSON()
+          );
+        });
+      });
+
+      it('renders dark icon when iconColor is set to business-color', () => {
+        const wrapper = renderStyles({ iconColor: 'business-color', bgTheme: 'none' });
+        assertStyleMatch(
+          {
+            color: baseTheme.colors.primary
+          },
+          wrapper.toJSON()
+        );
+
+        assertStyleMatch(
+          {
+            color: '#1E861E'
```

Comment:
we should test colors with theme objects instead of the direct output. A change in theme will lead to test failure.

Reply:
resolved in b526110